### PR TITLE
Revert "Revert "Generated iam policy updaters for terraform-google-conversion""

### PR DESCRIPTION
**Pull request by melinath**
Reverts GoogleCloudPlatform/magic-modules#4413

I am definitely feeling a bit ridiculous for this one. I waited on merging the last PR until I was feeling pretty sure that I could eliminate the remaining references to Config - but the more progress I made, the more issues I ran into. In particular, there are a number of references to private variables on the Config struct that would not be easy to eliminate. I didn't realize that variables on structs were also governed by the export rules.

So, TLDR, it seems like we should copy over these thousands of lines of code for now. Once the CAI conversion is done through the DCL we will be able to eliminate this, but in the meantime...

If this looks good I'll also do a force push on the terraform-google conversion master branch to eliminate a few large commits I made towards trying to use the provider config struct.

```release_note:none
```

### CODE REVIEW

**Review by slevenick — APPROVED** [review]
Can you fix the rubocop exceptions before merging?

Offenses:

provider/terraform_object_library.rb:175:55: W: Lint/UnusedMethodArgument: Unused method argument - generate_docs. If it's necessary, use _ or _generate_docs as an argument name to indicate that it won't be used.
    def generate_iam_policy(pwd, data, generate_code, generate_docs)
                                                      ^^^^^^^^^^^^^
provider/terraform_object_library.rb:191:7: C: Style/RedundantReturn: Redundant return detected.
      return
      ^^^^^^

**Comment by melinath** [comment]
😳  yes, yes I can fix those. Thank you for pointing them out! note to self to pay attention to run-rake-tests if I'm modifying ruby code.

**Comment by melinath** [comment]
Merging since this doesn't impact vcr tests
